### PR TITLE
#26 Eval footer

### LINKED ISSUE #26: Eval Footer
### User story
As a site user with any evaluation role, in order to build my trust using challenge.gov site, I would like to see consistent branding and styling from any page of the site.

**Acceptance criteria:**

- [x] Challenge.gov footer is displayed for logged-in users using Eval capability

### Definition of Done

**Doing (dev team)**
- [x] Code complete 
- [x] Code is organized appropriately
- [x] Any known trade offs are documented in the associated GH issue
- [x] Code is documented, modules, shared functions, etc.
- [x] Automated testing has been added or updated in response to changes in this PR
- [x] The feature is smoke tested to confirm it meets requirements
- [] ~~Database changes have been peer reviewed for index changes and performance bottlenecks~~ N/A
- [x] PR that changes or adds UI
  - [x] include a screenshot of the WAVE report for the altered pages
  - [x] Confirm changes were validated for mobile responsiveness
- [x] PR approved / Peer reviewed 
- [x] Security scans passed
- [x] Automate accessibility tests passed
- [x] Build process and deployment is automated and repeatable
- [ ] ~~Feature toggles if appropriate~~
- [x] Deploy to staging
- [x] Move card to testing column in the board
---
**Staging**
- [x] Accessibility tested (Marni)
  - [x]  Keyboard navigation
  - [x]  Focus confirmed
  - [x] Color contrast compliance
  - [x] Screen reader testing 
- [x] Usability testing: mobile and desktop (Tracy or Marni)
- [x] Cross browser testing - UI rendering is performant on below listed devices/browsers (Tracy or Marni)
  - [x] Windows/Chrome
  - [x] Windows/Edge
  - [x] Mac/Chrome
  - [x] Mac/Safari
  - [x] iOS/Safari
- [x] AC review (Renata)
- [ ] ~Deploy to production (production-like environment for eval capability) (dev team)~
- [x] Move to production column in the board
---
**Production**
- [ ] ~User and security documentation has been reviewed for necessary updates (Renata/Tracy/Dev team)~ 
- [x] PO / PM approved (Jarah or Renata)
- [x] AC is met and it works as expected (Jarah or Renata)
- [x] Move to done column in the board (Jarah or Renata)

### PR DESCRIPTION
**Pull request by bilalhankins**
Created footer layout
Added icons/logos to assets
Mobile responsiveness

Desktop View:
Screenshot 2024-08-13 at 4 14 52 PM

Mobile View:
Screenshot 2024-08-13 at 4 14 38 PM

https://github.com/GSA/Challenge_platform/issues/26

### CODE REVIEW

File: app/assets/uswds/_uswds-theme.scss [hunk 1]
```diff
@@ -1,13 +1,36 @@
-/*
-----------------------------------------
-USWDS with settings overrides
-----------------------------------------
-Uncomment the following lines and add a list of changed settings
-in the form $setting: value,
-----------------------------------------
-*/
```
Comment (stepchud) [inline app/assets/uswds/_uswds-theme.scss]: I'd like to keep this comment and maybe remove the "Uncomment the following lines" portion so it just says "Add a list of changed settings in the form $setting: value." for inline documentation

**Review by stepchud — COMMENTED** [review]
Added commit c877a07 which fixes the social icon SVG issues you were having

**Review by stepchud — COMMENTED** [review]
I was going to leave some minor UI feedback but it took me a while to play around with the styles and would take longer to explain everything so I added 5b3c862 instead. I think this is close enough to match the mockups. @bilalhankins can you test the responsiveness of this out in your local browser and see if you notice anything else that needs to be fixed?

**Comment by bilalhankins** [comment]
I tested and the responsiveness is way better than the changes i have locally. The only thing i saw was the space between the GSA logo and the right side of the screen on the mobile view. I tried to see if i could play with the css and eliminate the whitespace. I dont think it is too big of an issue tbh

**Comment by stepchud** [comment]
> I tested and the responsiveness is way better than the changes i have locally. The only thing i saw was the space between the GSA logo and the right side of the screen on the mobile view. I tried to see if i could play with the css and eliminate the whitespace. I dont think it is too big of an issue tbh

cool, I think it's fixed now

File: spec/requests/dashboard_request_spec.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+require "rails_helper"
+
+RSpec.describe "DashboardController" do
+  describe "GET /" do
+    before { get "/" }
+
+    it_behaves_like "a page with footer content"
```
Comment (stepchud) [inline spec/requests/dashboard_request_spec.rb:7]: ❤️

File: app/views/layouts/_footer.html.erb [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+      <div class="usa-footer__return-to-top">
+        <a href="#">Return to top</a>
+      </div>
+      <nav class="usa-footer__nav" aria-label="Footer navigation,,">
+        <div class="grid-row grid-gap-4 usa-footer__primary-content padding-top-2 mobile-lg:padding-top-0 usa-footer__primary-content--collapsible">
+          <div class="grid-col-5 mobile-lg:grid-col-6 desktop:grid-col-3">
+              <%# Col 1 %>
+              <ul class="usa-list usa-list--unstyled">
+                <li class="usa-footer__secondary-link">
+                  <a href="javascript:void(0);">About</a>
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]: add link
```suggestion
                  <a href="/about">About</a>
```

File: app/views/layouts/_footer.html.erb [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+                </li>
+                <li class="usa-footer__secondary-link">
+                  <a href="javascript:void(0);">Contact</a>
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
                  <a href="/contact">Contact</a>
```

File: app/views/layouts/_footer.html.erb [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+                </li>
+                <li class="usa-footer__secondary-link">
+                  <a href="javascript:void(0);">Terms of use</a>
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
                  <a href="/terms">Terms of use</a>
```

File: app/views/layouts/_footer.html.erb [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+                </li>
+                <li class="usa-footer__secondary-link">
+                  <a href="javascript:void(0);">Privacy Policy</a>
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
                  <a href="/privacy-policy">Privacy Policy</a>
```

File: app/views/layouts/_footer.html.erb [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+                </li>
+              </ul>
+          </div>
+          <div class="grid-col-5 display-block desktop:display-none mobile-lg:grid-col-6">
+          </div>
+          <div class="grid-col-5 mobile-lg:grid-col-6 desktop:grid-col-3">
+              <%# Col 2 %>
+              <ul class="usa-list usa-list--unstyled">
+                <li class="usa-footer__secondary-link">
+                  <a href="javascript:void(0);">Innovator Resources</a>
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
                  <a href="/resources">Innovator Resources</a>
```

File: app/views/layouts/_footer.html.erb [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+                </li>
+                <li class="usa-footer__secondary-link">
+                  <a href="javascript:void(0);">Agency Resources</a>
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
                  <a href="/resources-managers">Agency Resources</a>
```

File: app/views/layouts/_footer.html.erb [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+                </li>
+                <li class="usa-footer__secondary-link">
+                  <a href="javascript:void(0);">Archived Challenges</a>
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
                  <a href="/?state=archived">Archived Challenges</a>
```

File: app/views/layouts/_footer.html.erb [hunk 10]
```diff
@@ -0,0 +1,121 @@
...
+                </li>
+                <li class="usa-footer__secondary-link">
+                  <a href="javascript:void(0);">Active Challenges</a>
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
                  <a href="/#active-challenges">Active Challenges</a>
```

File: app/views/layouts/_footer.html.erb [hunk 11]
```diff
@@ -0,0 +1,121 @@
...
+                </li>
+              </ul>
+          </div>
+          <div class="grid-col-5 mobile-lg:grid-col-6 desktop:grid-col-3">
+              <%# Col 3 %>
+              <ul class="usa-list usa-list--unstyled">
+                <li class="usa-footer__secondary-link">
+                  <a href="javascript:void(0);">GSA Privacy Policy</a>
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
                  <a href="https://www.gsa.gov/website-information/website-policies#privacy">GSA Privacy Policy</a>
```

File: app/views/layouts/_footer.html.erb [hunk 12]
```diff
@@ -0,0 +1,121 @@
...
+                </li>
+                <li class="usa-footer__secondary-link">
+                  <a href="javascript:void(0);">Accessibility</a>
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
                  <a href="https://www.gsa.gov/website-information/accessibility-aids">Accessibility</a>
```

File: app/views/layouts/_footer.html.erb [hunk 13]
```diff
@@ -0,0 +1,121 @@
...
+                </li>
+                <li class="usa-footer__secondary-link">
+                  <a href="javascript:void(0);">FOIA</a>
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
                  <a href="https://www.gsa.gov/reference/freedom-of-information-act-foia">FOIA</a>
```

File: app/views/layouts/_footer.html.erb [hunk 14]
```diff
@@ -0,0 +1,121 @@
...
+                </li>
+                <li class="usa-footer__secondary-link">
+                  <a href="javascript:void(0);">USA.gov</a>
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
                  <a href="https://www.usa.gov/">USA.gov</a>
```

File: app/views/layouts/_footer.html.erb [hunk 15]
```diff
@@ -0,0 +1,121 @@
...
+            <img class="usa-footer__logo-img" src="<%= image_path('challenge-logo-white-orange.svg') %>" title="Challenge Gov wordmark" alt="Challenge Gov wordmark in footer" style="width: 220px; height: auto;" />
+          </div>
+          <div class="order-2 tablet:order-1">
+            <img class="usa-footer__logo-img" src="<%= image_path('challenge-logo-square-white.svg') %>" title="GSA Star Mark" alt="GSA Star Mark" style="width: 60px; height: auto;" />
+          </div>
+        </div>
+        <div class="usa-footer__contact-links mobile-lg:grid-col-12 tablet:grid-col-6 margin-y-1 footer-white">
+          <div class="usa-footer__social-links flex-justify-end grid-row grid-gap-1">
+            <div class="grid-col-auto">
+              <a class="usa-social-link" href="javascript:void(0);">
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
              <a class="usa-social-link" href="https://www.twitter.com/ChallengeGov">
```

File: app/views/layouts/_footer.html.erb [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+                <%= image_tag('images/usa-icons/twitter.svg', class: "usa-social-link__icon", alt: "Twitter") %>
+              </a>
+            </div>
+            <div class="grid-col-auto">
+              <a class="usa-social-link" href="javascript:void(0);">
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
              <a class="usa-social-link" href="https://www.facebook.com/ChallengeGov">
```

File: app/views/layouts/_footer.html.erb [hunk 17]
```diff
@@ -0,0 +1,121 @@
...
+                <%= image_tag('images/usa-icons/facebook.svg', class: "usa-social-link__icon", alt: "Facebook") %>
+              </a>
+            </div>
+            <div class="grid-col-auto">
+              <a class="usa-social-link" href="javascript:void(0);">
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
              <a class="usa-social-link" href="https://www.youtube.com/@challenge-gov">
```

File: app/views/layouts/_footer.html.erb [hunk 18]
```diff
@@ -0,0 +1,121 @@
...
+                <%= image_tag('images/usa-icons/youtube.svg', class: "usa-social-link__icon", alt: "YouTube") %>
+              </a>
+            </div>
+            <div class="grid-col-auto">
+              <a class="usa-social-link" href="javascript:void(0);">
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
              <a class="usa-social-link" href="https://www.linkedin.com/company/challengegov/">
```

File: app/views/layouts/_footer.html.erb [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+                <%= image_tag('images/usa-icons/linkedin.svg', class: "usa-social-link__icon", alt: "LinkedIn") %>
+              </a>
+            </div>
+            <div class="grid-col-auto">
+              <a class="usa-social-link" href="javascript:void(0);">
```
Comment (stepchud) [inline app/views/layouts/_footer.html.erb]:
```suggestion
              <a class="usa-social-link" href="https://github.com/GSA/Challenge_gov">
```

**Review by stepchud — CHANGES_REQUESTED** [review]
the internal app links won't work yet but the external ones will. we can go ahead and add all the links now while we build those internal pages. 
copied these from the production site at https://www.challenge.gov

**Review by stepchud — APPROVED** [review]
think we're g2g 🎉

**Comment by stepchud** [comment]
WAVE report:
Screenshot 2024-08-16 at 9 26 05 AM

**Pull request merged**